.gitlab-ci-black-format-code.yml

```diff
@@ -0,0 +1,73 @@
+# .gitlab-ci-black-format-code.yml
+#
+# This gitlab-ci file runs the Black code formatter https://github.com/psf/black
+# on the Python3 code in the repository, and fails the pipeline if the code is
+# not formatted according to standard Black formatting rules.
+#
+# Add it to the project in hand through Gitlab's include functionality
+# 
+# include:
+#   - project: 'Northern.tech/Mender/mendertesting'
+#     file: '.gitlab-ci-black-format-code.yml'
+#
+#
+# Some things to consider before including this file. The Black code formatter
+# accepts a project-local configuration file
+# (https://github.com/psf/black#pyprojecttoml). This enables project specific
+# configuration of the Black tool.
+#
+# The recommended parameters to specify are:
+# * The Python version
+# * Include and exlude files
+#
+#
+# Example 'pyproject.toml' file:
+#
+#[tool.black]
+# target-version = ['py38']
+# include = '\.pyi?$' # Regular expression
+# exclude = '''       # Regular expression
+# (
+#   /(
+#       \.eggs         # exclude a few common directories in the
+#     | \.git          # root of the project
+#     | \.hg
+#     | \.mypy_cache
+#     | \.tox
+#     | \.venv
+#     | _build
+#     | buck-out
+#     | build
+#     | dist
+#   )/
+#   | foo.py           # also separately exclude a file named foo.py in
+#                      # the root of the project
+# )
+# '''
+# The tools also ignores all files found in .gitignore.
+
+stages:
+  - test
+
+test:check-python3-formatting:
+  image: alpine
+  stage: test
+  before_script:
+    # Install dependencies
+    - apk add --no-cache git bash python3
+    - pip3 install --upgrade pip setuptools
+    - pip3 install black # TODO -- Should we freeze the version here (?)
+    # Rename the branch we're on, so that it's not in the way for the
+    # subsequent fetch. It's ok if this fails, it just means we're not on any
+    # branch.
+    - git branch -m temp-branch || true
+    # Git trick: Fetch directly into our local branches instead of remote
+    # branches.
+    - git fetch origin 'refs/heads/*:refs/heads/*'
+    # Get last remaining tags, if any.
+    - git fetch --tags origin
+    - git clone http://github.com/mendersoftware/mendertesting /tmp/mendertesting
+  script:
+    # Check that the python code is formattet
+    # If no arguments are provided, the script will check the current directory.
+    - /tmp/mendertesting/check_python_format "$CHECK_PYTHON_FORMAT_ARGS"
```

Comment:
These arguments are not documented anywhere. This is something that the caller pipelines can use, so I suggest we add it to the above documentation.